src/components/SearchBar/index.jsx

```diff
@@ -2,13 +2,30 @@ import React from 'react';
 import { SearchField } from '@edx/paragon';
 import qs from 'query-string';
 import PropTypes from 'prop-types';
+import { getCurrentPage } from '../../components/SiteHeader';
 
+const SEARCH_PATH = '/search/';
 
 class SearchBar extends React.Component {
   constructor(props) {
     super(props);
     this.search = this.search.bind(this);
   }
+
+  getJournalIdForSearch() {
+    // If displaying the index page, search across all journals
+    // by using journalId=0
+    if (getCurrentPage(null, null, this.props.journalIndexId) === this.props.journalIndexId) {
```

Comment:
I'm not able to test it on `journalIndexId `. Can you guide me how i can do that?  If I try to access 
```
http://localhost:1991/{journalIndexId}
```
it redirects me to 
```
http://localhost:1991/{journalIndexId}/{pageId}
```

Reply:
if you have two journals defined and you go to localhost:1991 it will display the index page and set journalId = 0 (via `resetJournal()` method)
This will take you to a journal about page `http://localhost:1991/{journalAboutPageId}` but note that `journalId` is different than `journalAboutPageId` and you cannot directly reference a `journalId` from the url